rewrite-java/src/main/java/org/openrewrite/java/search/SemanticallyEqual.java

```diff
@@ -1250,10 +1187,6 @@ public J.Try.Resource visitTryResource(J.Try.Resource tryResource, J j) {
                 }
 
                 J.Try.Resource compareTo = (J.Try.Resource) j;
-                if (tryResource.isTerminatedWithSemicolon() != compareTo.isTerminatedWithSemicolon()) {
-                    isEqual.set(false);
-                    return tryResource;
-                }
```

Comment:
I completely removed this if block, as it appears to be wrong to check for this in the `SemanticallyEqual` class.

Reply:
Agree, that is syntactic equality rather than semantic equality